File: versioncontrol/src/main/java/ir/ayantech/versioncontrol/CallVCApi.java
package ir.ayantech.versioncontrol;


import ir.ayantech.versioncontrol.api.VCResponseStatus;
import ir.ayantech.versioncontrol.model.VCInputModel;

/**
 * Created by shadoWalker on 5/9/17.
 */

public interface CallVCApi<RequestModel extends VCInputModel> {
    void callApi(VCResponseStatus status, RequestModel inputModel);
}

File: versioncontrol/src/main/java/ir/ayantech/versioncontrol/CallVCApi.kt
package ir.ayantech.versioncontrol

import ir.ayantech.versioncontrol.api.VCResponseStatus
import ir.ayantech.versioncontrol.model.VCInputModel

interface CallVCApi<RequestModel : VCInputModel> {
    fun callApi(status: VCResponseStatus, inputModel: RequestModel)
}
